### PR TITLE
Edge identity service foundation

### PR DESCRIPTION
**Pull request by GerardoGR**
This PR introduces a new section called: `Edges` where we create the edges (highly-coupled with the spire join-tokens) to be able to register the FL edge part. 

fl-suite-edges

---

fl-suite-edge-detail

This implements the following:

- Creates, shows and deletes an Edge (join-token)
- When showing an edge, it presents an `FLEdge` CRD (WIP: If implemented in this same PR it will be too unwieldy) to deploy the edge kubernetes compoenents.
- The edge-identity service deployed inside the spire-server pod to access the grpc unix-socket for admin access (e.g create join-tokens).
- Storing the state of the edge in an embedded SQLite database. This is on purpose because the feature is not production ready.

ToDo (but meant to be implemented in following PRs):
- [ ] Introduce new `FLEdge` CRD to provision the edge (e.g spire-agent, `FLOrchestrator` CRD, etc.)
- [ ] Persist the `Edge` data currently stored in SQLite. An idea could be to store this in MySQL because it is currently deployed by kubeflow.
- [ ] Reconsider the UI implementation: Currently it was done by reusing the common crud-webapp libraries provided by kubeflow (Angular 12, Material UI, etc), but this could be implemented with another stack.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
+[submodule "components/vendor/kubeflow-manifests"]
+	path = components/vendor/kubeflow-manifests
+	url = git@github.com:kubeflow/manifests.git
```
Comment (nfnt) [inline .gitmodules]: Let's either have Git or HTTPS URLs for the submodules, but not both at the same time. I would prefer HTTPS URLs, as one doesn't have to add any SSH certs in Github to be able to use them.

Comment (GerardoGR) [reply .gitmodules]: That was a typo, but thanks

File: components/edge-identity/backend/apps/default/schema.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE IF NOT EXISTS Edge (
+	edge_id INTEGER PRIMARY KEY,
+	name TEXT NOT NULL,
+	namespace TEXT NOT NULL,
+	created_at TEXT NOT NULL,
+	join_token TEXT NOT NULL
```
Comment (orlandohohmeier) [inline components/edge-identity/backend/apps/default/schema.sql]: Both for the schema as well as the model please keep in mind that we will need to support different types of edges i.e. ones using "join tokens" to authenticate but also ones that will use different means like pre-shared certs. Thus I'd use a table to store the edge identifier and type plus dedicated tables and specific models for the respective edge type details like the join token.

**Review by orlandohohmeier — CHANGES_REQUESTED** [review]

File: components/edge-identity/backend/apps/default/schema.sql (hunk 2 above)
Comment (GerardoGR) [reply components/edge-identity/backend/apps/default/schema.sql]: Makes sense. I've added a new `JoinTokenAuth` table that references each `Edge`.

File: components/edge-identity/backend/apps/common/__init__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import kubeflow.kubeflow.crud_backend as base
```
Comment (orlandohohmeier) [inline components/edge-identity/backend/apps/common/__init__.py]: I might be missing something, but as it seems that the _common_ app isn't used, please remove all the respective files.

File: components/edge-identity/backend/apps/default/db.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import click
+from flask import current_app, g
+from flask.cli import with_appcontext
+
+DATABASE = "edge-identity.db"
+
+
+def get_db():
+    if "db" not in g:
+        g.db = sqlite3.connect(DATABASE, detect_types=sqlite3.PARSE_DECLTYPES)
```
Comment (orlandohohmeier) [inline components/edge-identity/backend/apps/default/db.py:13]: 🤔 Not relevant for this PR, but I'm wondering what it takes to change the storage backend... thinking that we might be able to utilize `etcd` and have this service just create the current resource plus an operator which in turn configures spire and so on...

File: components/edge-identity/backend/apps/common/__init__.py (hunk 3 above)
Comment (GerardoGR) [reply components/edge-identity/backend/apps/common/__init__.py]: Good catch!. I'll remove them. Thanks!

File: components/edge-identity/backend/apps/default/db.py (hunk 4 above)
Comment (GerardoGR) [reply components/edge-identity/backend/apps/default/db.py:13]: It's an approach worth checking. Let's discuss this offline and comeup with follow-ups.

File: components/edge-identity/backend/apps/common/__init__.py (hunk 3 above)
Comment (GerardoGR) [reply components/edge-identity/backend/apps/common/__init__.py]: common app removed in: https://github.com/katulu-io/fl-suite/pull/15/commits/5fb84e2931d0a2d95263e9d4f8f009625297548a

**Review by orlandohohmeier — APPROVED** [review]

**Pull request merged**